Docs: Update expression documentation to mention no data (#61934)
grafana/grafana: docs/sources/panels-visualizations/query-transform-data/expression-queries/index.md

1. Write the expression.
1. Click **Apply**.

## Special cases

> When data source returns no data points the expression engine produces special result `NoData`.  All operations with this type of result result in NoData. If for example, the request contains two queries to data source that are merged by math expression, NoData returned by at least one of data source queries will casue the result of the math expression and all subsequent chain of expressions to be NoData.

Reviewer: > When data source returns no data points the expression engine produces special result `NoData`.

I think it might be more accurate to say "no data items", or perhaps "no series or numbers", as there could be a Series with no datapoints in it, and I don't think this gets classified as "No Data".

Ref: https://github.com/grafana/grafana-plugin-sdk-go/tree/main/data/contract_docs#no-data-and-empty

> When data source returns no series or numbers the expression engine produces special result `NoData`.  All operations with this type of result result in NoData. If for example, the request contains two queries to data source that are merged by math expression, NoData returned by at least one of data source queries will cause the result of the math expression and all subsequent chain of expressions to be NoData.

Reviewer 2: ```suggestion
When any queried data source returns no series or numbers, the expression engine returns `NoData`.  For example, if a request contains two data source queries that are merged by an expression, if `NoData` is returned by at least one of the data source queries, then the returned result for the entire query is `NoData`.
```

> NoData result allows [Grafana Alerting]({{< relref "../../../alerting/" >}}) to handle this case in a special way. Read more about [No data and error handling]({{< relref "../../../alerting/alerting-rules/create-grafana-managed-rule/#no-data-and-error-handling" >}})

Reviewer 2: ```suggestion
For more information about how [Grafana Alerting]({{< relref "../../../alerting/" >}}) processes `NoData` results, refer to [No data and error handling]({{< relref "../../../alerting/alerting-rules/create-grafana-managed-rule/#no-data-and-error-handling" >}}).
```
